src/build_tools.rs

```diff
@@ -248,3 +248,33 @@ pub fn safe_repr(v: &PyAny) -> Cow<str> {
         "<unprintable object>".into()
     }
 }
+
+#[derive(Debug, Clone)]
+pub(crate) enum ExtraBehavior {
```

Comment:
since this isn't only used at build time, I'd rather this went in `src/validators/typed_dict.rs`.

Reply:
It gets used by dataclasses and typed dict serializer as well

Reply:
I still think better on balance to put it in one of the `validators/*` files, that's what we do either other similar reusable structs/enums.